src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,682 @@
+        let plane_shape_offset = Var::<Distance<Pixels>>::from(&crop_plane_pos - width * 0.5);
+        let crop_shape         = HalfPlane();
+        let crop_shape         = crop_shape.rotate(plane_rotation_angle);
+        let crop_shape         = crop_shape.translate_x(plane_shape_offset);
+
+        crop_shape.into()
+    }
+
+    ensogl::define_shape_system! {
+        (style:Style, grow:f32, index:f32, port_num:f32, opacity:f32, padding_left:f32,
+        padding_right:f32) {
+            let overall_width  : Var<Distance<Pixels>> = "input_size.x".into();
+            let overall_height : Var<Distance<Pixels>> = "input_size.y".into();
+
+            let base_shape_data = port_base_shape_data(&overall_width, &overall_height, &grow);
+            let BaseShapeData{ port_area,hover_area,radius,width } = base_shape_data;
+
+            let left_shape_crop  = compute_crop_plane(&index,
+                                                      &port_num,
+                                                      &width.clone().into(),
+                                                      &radius.clone().into(),
+                                                      &padding_left);
+            let right_shape_crop = compute_crop_plane(&(Var::<f32>::from(1.0) + &index),
+                                                      &port_num,
+                                                      &width.clone().into(),
+                                                      &radius.clone().into(),
+                                                      &padding_right);
+
+            let port_area  = port_area.difference(&left_shape_crop);
+            let port_area  = port_area.intersection(&right_shape_crop);
+
+            let left_hover_crop  = compute_crop_plane(&index, &port_num, &width.clone().into(),
+                                                      &radius.clone().into(), &0.0.into());
+            let right_hover_crop = compute_crop_plane(&(Var::<f32>::from(1.0) + &index), &port_num,
+                                                      &width.into(), &radius.into(), &0.0.into());
+
+            let hover_area = hover_area.difference(&left_hover_crop);
+            let hover_area = hover_area.intersection(&right_hover_crop);
+            let hover_area = hover_area.fill(color::Rgba::new(0.0,0.0,0.0,0.000_001));
+
+            // FIXME: Use colour from style and apply transparency there.
+            let color     = Var::<color::Rgba>::from("srgba(0.25,0.58,0.91,input_opacity)");
+            let port_area = port_area.fill(color);
+
+            (port_area + hover_area).into()
+        }
+    }
+
+    impl PortShapeApi for Shape {
+        fn set_grow(&self, grow_value:f32) {
+            self.grow.set(grow_value)
+        }
+
+        fn set_opacity(&self, opacity:f32) {
+            self.opacity.set(opacity)
+        }
+    }
+}
+
+/// Implements a simplified version of the multi_port_area::Shape shape for the case where there is
```

Comment:
Lets create a big section here to divide these codes. They are pretty big and it would be cool to discover it while fast scrolling :)